tests/generative/test_query_model.py

```diff
@@ -45,14 +42,22 @@
 
 
 @pytest.fixture(scope="session")
-def query_engines():
+def query_engines(request):
+    # This is a slight abuse of pytest's internals but it achieves what we want, which
+    # is to get all the values of a parametrized fixture in one go. Usually we want to
+    # run tests against each query engine one at a time, but here we want to run the
+    # same query against multiple query engines together so we can compare the results.
```

Comment:
It does seem a bit odd to use the fixture as a catalogue of engines. Is there a possibility instead of factoring the list of named engines out of the fixture and accessing that instead?

Reply:
Oh you're quite right. We can do this without any shenanigans at all — can't we? — with a bit of refactoring. I got carried away with my own terribleness.